### PR TITLE
ENH remove duplication of ScatteringND classes

### LINKED ISSUE #488: Remove redundant loading of frontends
Do we wanna define a subclass `ScatteringBase` in the main `kymatio` folder, from which `Scattering*D` subclass inherits from, and all we do here is init the `ScatteringBase`? 

@lostanlen @janden your thoughts are also appreciated.

_Originally posted by @MuawizChaudhary in https://github.com/kymatio/kymatio/diffs_

Right now, the dynamic loading of frontends (i.e., the `self.__class__` trick) is replicated for each of the `Scattering*D` classes. This should be moved into a shared superclass or some auxiliary function.

### PR DESCRIPTION
**Pull request by edouardoyallon**
Closes #488

### CODE REVIEW

File: kymatio/frontend/__init__.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+            self.__class__ = getattr(module, class_name)
+            self.__init__(*args, **kwargs)
+        except Exception as e:
+            raise e from RuntimeError('\nThe frontend \'' + frontend + '\' could not be correctly imported.')
+
+        logging.info('The ' + self.name + ' frontend ' + frontend + ' was imported.')
+
+
+__all__ = ['Scattering']
```
Comment (janden) [inline kymatio/frontend/__init__.py]: Add newline.

File: kymatio/numpy.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+from scattering1d.frontend.numpy_frontend import ScatteringNumPy1D as Scattering1D
+from scattering2d.frontend.numpy_frontend import ScatteringNumPy2D as Scattering2D
+
+__all__ = ['Scattering1D', 'Scattering2D']
```
Comment (janden) [inline kymatio/numpy.py]: Add newline.

File: kymatio/numpy.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+from scattering1d.frontend.numpy_frontend import ScatteringNumPy1D as Scattering1D
```
Comment (janden) [inline kymatio/numpy.py]: Use relative imports.

File: kymatio/frontend/__init__.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        self.name = kwargs['name']
+        self.class_name = kwargs['class_name']
+        kwargs.pop('name')
+        kwargs.pop('class_name')
+
+        frontend_suffixes = {'torch' : 'Torch', 'numpy' : 'NumPy', 'tensorflow'
+                : 'TensorFlow'}
+        if 'frontend' not in kwargs:
+            warnings.warn("Torch frontend is currently the default, but NumPy will become the default in the next"
+                          " version.", PendingDeprecationWarning)
```
Comment (janden) [inline kymatio/frontend/__init__.py]: `DeprecationWarning`

File: kymatio/tensorflow.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+from scattering1d.frontend.tensorflow_frontend import ScatteringTensorFlow1D as Scattering1D
```
Comment (janden) [inline kymatio/tensorflow.py]: Relative imports.

File: kymatio/tensorflow.py [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+from scattering2d.frontend.tensorflow_frontend import ScatteringTensorFlow2D as Scattering2D
+
+__all__ = ['Scattering1D', 'Scattering2D']
```
Comment (janden) [inline kymatio/tensorflow.py]: Add newline.

File: kymatio/torch.py [hunk 7]
```diff
@@ -0,0 +1,4 @@
+from scattering1d.frontend.torch_frontend import ScatteringTorch1D as Scattering1D
```
Comment (janden) [inline kymatio/torch.py]: Relative imports.

File: kymatio/torch.py [hunk 8]
```diff
@@ -0,0 +1,4 @@
...
+from scattering2d.frontend.torch_frontend import ScatteringTorch2D as Scattering2D
+
+__all__ = ['Scattering1D', 'Scattering2D']
```
Comment (janden) [inline kymatio/torch.py]: Add newline.

**Comment by edouardoyallon** [comment]
It's done.

**Comment by janden** [comment]
Ok :+1:.

**Pull request merged**